### PR TITLE
Use TOML serializer from nixpkgs

### PR DESCRIPTION
**Pull request by Patryk27**
Requires https://github.com/nix-community/naersk/pull/288.
Closes https://github.com/nix-community/naersk/issues/263.

### CODE REVIEW

**Review by nmattia — DISMISSED** [review]
Thanks! Didn't know this existed.

**Comment by Patryk27** [comment]
I've rebased just now.

> Thanks! Didn't know this existed.

Yeah, the older version in nixpkgs was kinda bad (e.g. didn't support stuff like `[targets."cfg(...)"]`), but the current one looks pretty complete :smile:

**Comment by Patryk27** [comment]
Ouch, something's failing on the CI - I'll take a look later 🤨

**Comment by Patryk27** [comment]
Alright, the issue is that older nixpkgs' (< 23) `lib.formats.toml` doesn't properly handle cases like `[target."cfg(\"something\")"]`, which means that we can't rely on it there.

I've adjusted the logic to use our custom serializer (as it was) for older nixpkgs and rely on the nixpkgs' serializer only when nixpkgs are recent enough, with the intention of getting rid of our custom serializer when nixpkgs v23+ get more widely adopted.

**Comment by Patryk27** [comment]
Huh, it's still failing, but the tests succeed on my Linux machine 🤨 -- lemme check on another one

**Comment by Patryk27** [comment]
Alrighty, done!

File: nix/sources.json [hunk 1]
```diff
@@ -60,15 +60,15 @@
         "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
     },
     "nixpkgs": {
-        "branch": "master",
+        "branch": "nixpkgs-unstable",
```
Comment (nmattia) [inline nix/sources.json:63]: any chance we can stay on `master`?

**Review by nmattia — DISMISSED** [review]
Thanks a lot!

File: nix/sources.json (hunk 1 above)
Comment (Patryk27) [reply nix/sources.json]: Sure, done 🙂

**Comment by Patryk27** [comment]
hmm, analyzing failing tests 🔍

**Comment by Patryk27** [comment]
Alright, got it!

I think https://github.com/nix-community/naersk/pull/299 caused us to (randomly) bump into https://github.com/rust-lang/cargo/pull/11188 where tl;dr it looks like when both crates.io & Git dependencies are packed together, as we do now:

https://github.com/nix-community/naersk/blob/d9a33d69a9c421d64c8d925428864e93be895dcc/build.nix#L91

... Cargo (sometimes, randomly) decides that Git-based dependencies must have checksums specified, which we don't do for them, since we don't know that checksum:

https://github.com/nix-community/naersk/blob/d9a33d69a9c421d64c8d925428864e93be895dcc/build.nix#L466

Creating two separate sources (one for crates.io and one for Git dependencies), as I've done in the recent commit just now, seems to avoid the issue (I've checked on aarch64-darwin, x86_64-linux NixOS + x86_64-linux Ubuntu, and everything passed on the first time now).

**Review by nmattia — APPROVED** [review]
Thanks for all the work!

**Pull request merged**